coderd/database/queries/workspaceagentportshare.sql

```diff
@@ -0,0 +1,11 @@
+-- name: InsertWorkspaceAgentPortShare :one
+INSERT INTO workspace_agent_port_share (workspace_id, agent_name, port, share_level) VALUES ($1, $2, $3, $4) RETURNING *;
+
+-- name: GetWorkspaceAgentPortShare :one
+SELECT * FROM workspace_agent_port_share WHERE workspace_id = $1 AND agent_name = $2 AND port = $3;
```

Comment:
I'm guessing you're going to add a list query+endpoint when you do the FE?

Reply:
yup, will probably be added to the workspaceagent endpoint alongside the detected ports